src/rgw/rgw_datalog.cc

```diff
@@ -70,12 +72,10 @@ void rgw_data_change_log_entry::decode_json(JSONObj *obj) {
 
 class RGWDataChangesOmap final : public RGWDataChangesBE {
   using centries = std::list<cls_log_entry>;
-  RGWSI_Cls& cls;
```

Comment:
Not exactly. Since RGWDataChangesLog has to have an IoCtx to pass to `log_acquire_backing` and friends, it seemed silly for the backends to have to make their own instead of it just being passed into them. Especially since `svc_cls` just created and dropped an IoCtx with every call, just having one and using it seemed preferable.